### PR TITLE
[CFM-1257, CFM-1258] Add promote / demote icon.

### PR DESCRIPTION
**Pull request by nirgn975**
https://issuetracker.amplexor.com/jira/browse/CFM-1257
https://issuetracker.amplexor.com/jira/browse/CFM-1258

### CODE REVIEW

**Comment by nirgn975** [comment]
capacity4more

capacity4more

capacity4more

**Comment by nirgn975** [comment]
@ordavidil Ready for review.

**Comment by ordavidil** [comment]
@nirgn975 
please update the PR as discussed on the call with Diane

**Comment by nirgn975** [comment]
- [ ] Create a block in the right side of every activity.
- [ ] Remove the text of the flag.
- [ ] Add an `X` on the pin icon when it's pinned (to unpin it).
- [ ] Check that only `power users` see the block option.
- [ ] Make sure we can promote couple of item at the same time.

**Comment by ordavidil** [comment]
@nirgn975 
looks like yes, besides the last one:
> Make sure we can promote couple of item at the same time.

I think we can promote only one at once.

**Comment by nirgn975** [comment]
@ordavidil Yep, my bad. I meant to write `can't`.

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_activity_stream.scss [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
     border-bottom: 1px dashed $color-gray-light;
+    position: relative;
+
+    .block-action {
+      position: absolute;
+      right: 0;
+      top: 20px;
+
+      i {
+      transform: rotate(0deg);
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_activity_stream.scss]: fix indentation

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 2]
```diff
@@ -1559,3 +1559,58 @@ function c4m_message_get_user_last_activity($uid) {
...
+  $node = $wrapper->field_node->value();
+
+  return flag_create_link('c4m_og_content_highlight', $node->nid);
+}
+
+/**
+ * Implements hook_block_info().
+ */
+function c4m_message_block_info() {
+  $blocks['c4m_group_activity_actions_block'] = array(
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: Change the block name to `c4m_message_power_user_actions`

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 3]
```diff
@@ -1559,3 +1559,58 @@ function c4m_message_get_user_last_activity($uid) {
...
+
+  if ($delta != 'c4m_group_activity_actions_block') {
+    return $block;
+  }
+
+  if (!$message) {
+    return $block;
+  }
+
+  $tag['element'] = array(
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: Check if the current user is power user,  if not then return early.

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 4]
```diff
@@ -1559,3 +1559,58 @@ function c4m_message_get_user_last_activity($uid) {
...
+    '#tag' => 'div',
+    '#value' => c4m_message_format_promote($message),
+    '#attributes' => array(
+      'class' => array('block-action'),
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: rename to `c4m-og-power-user-actions-block`

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_activity_stream.scss [hunk 5]
```diff
@@ -17,12 +28,23 @@
...
     margin-top: 25px;
   }
-  
+
   .message-summary {
     padding: 0;
   }
 }
 
+.pane-og-dashboard {
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_activity_stream.scss]: This should be under `.c4m-og-power-user-actions-block`

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/base/_mixins.scss [hunk 6]
```diff
@@ -101,3 +101,9 @@
     top: 0;
   }
 }
+
+@mixin align_flag {
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/base/_mixins.scss]: =>
```scss
@mixin rotate_promote_flag {
```

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_activity_stream.scss (hunk 5 above)
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_activity_stream.scss]: @nirgn975 
if I remember correct, we have this flag also on the group dashboard.
please make sure it doesn't effect this one as well...

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.module [hunk 7]
```diff
@@ -293,10 +293,19 @@ function c4m_features_og_group_dashboard_preprocess(&$variables, $hook) {
     return;
   }
 
+  $item = menu_get_item();
+
+  // Check that the flag is on a page, and not on the activity_group.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_group_dashboard/c4m_features_og_group_dashboard.module]: Fix the comment.

**Pull request merged**